libsolc/CMakeLists.txt

```diff
@@ -0,0 +1,9 @@
+if (EMSCRIPTEN)
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s EXPORTED_FUNCTIONS='[\"_compileJSON\",\"_license\",\"_version\",\"_compileJSONMulti\",\"_compileJSONCallback\",\"_compileStandard\"]' -s RESERVED_FUNCTION_POINTERS=20")
+	add_executable(soljson libsolc.cpp)
+	target_link_libraries(soljson PRIVATE solidity)
+else()
+	add_library(libsolc SHARED libsolc.cpp)
```

Comment:
The `SHARED` part could be skipped for this PR.

Reply:
Why do you need it to be a shared library.

Reply:
Ultimate goal of #2864.

Reply:
So let's keep it here now.